File: 2.1. Two Pointers/1. Valid Palindrome/dory.py
# Runtime 48ms
# Beats 39.01%

# Memory 18.12MB
# Beats 17.81%
import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        def remove_non_alphabet(t):
            return re.sub(r'[^a-zA-Z0-9]', '', t)
        buffer = remove_non_alphabet(s).lower()
        print(buffer)
        return buffer == buffer[::-1]

File: 2.1. Two Pointers/1. Valid Palindrome/dory.cs
// Runtime 35ms
// Beats 23.99%

// Memory 46.18MB
// Beats 34.84%

public class Solution {
    public bool IsPalindrome(string s) {
        string result = new string(s.Where(c => char.IsLetter(c)).ToArray()).ToLower();

        // If considering number 
        // string result = new string(s.Where(c => char.IsLetterOrDigit(c)).ToArray()).ToLowerInvariant();

        if(result.Length <= 1) {
            return true;
        }
        int start = 0;
        int end = result.Length - 1;
        while(start < end) {
            if(result[start] == result[end]) {
                start++;
                end--;
            } else {
                return false;
            }
        }
        return true;
    }
}